packages/dynamite/dynamite/lib/src/models/openapi/schema.dart

```diff
@@ -47,7 +49,36 @@ abstract class Schema implements Built<Schema, SchemaBuilder> {
 
   Schema? get items;
 
-  Schema? get additionalProperties;
+  /// Value can be boolean or object.
+  ///
+  /// Inline or referenced schema MUST be of a Schema Object and not a standard JSON Schema.
+  /// Consistent with JSON Schema, additionalProperties defaults to true.
+  ///
+  /// See:
+  ///   * https://github.com/OAI/OpenAPI-Specification/issues/668
+  ///   * https://json-schema.org/understanding-json-schema/reference/object#additionalproperties
+  @memoized
+  Schema get parsedAdditionalProperties {
+    final properties = additionalProperties;
+
+    if (properties.isBool) {
+      if (properties.asBool) {
+        return Schema();
+      } else {
+        throw UnsupportedError('Matching that a value does not match a schema is not yet supported.');
+      }
+    }
+
+    return openapi.serializers.deserializeWith(
+      serializer,
+      properties.value,
+    )!;
+  }
+
+  /// Use [parsedAdditionalProperties] instead.
+  @visibleForOverriding
+  @BuiltValueField(wireName: 'additionalProperties')
+  JsonObject get additionalProperties;
```

Comment:
Wire name is not needed. IMO it would be better to name this getter something like `rawAdditionalProperties` (and keep the wire name) and then call the `parsedAdditionalProperties` just `additionalProperties`.